app/src/main/java/at/roteskreuz/stopcorona/screens/reporting/reportStatus/ReportingStatusViewModel.kt

```diff
@@ -1,38 +1,107 @@
 package at.roteskreuz.stopcorona.screens.reporting.reportStatus
 
 import at.roteskreuz.stopcorona.model.entities.infection.message.MessageType
+import at.roteskreuz.stopcorona.model.exceptions.SilentError
 import at.roteskreuz.stopcorona.model.repositories.AgreementData
+import at.roteskreuz.stopcorona.model.repositories.ExposureNotificationRepository
 import at.roteskreuz.stopcorona.model.repositories.QuarantineRepository
 import at.roteskreuz.stopcorona.model.repositories.ReportingRepository
 import at.roteskreuz.stopcorona.skeleton.core.model.helpers.AppDispatchers
 import at.roteskreuz.stopcorona.skeleton.core.model.helpers.DataState
 import at.roteskreuz.stopcorona.skeleton.core.model.helpers.DataStateObserver
+import at.roteskreuz.stopcorona.skeleton.core.model.helpers.State
 import at.roteskreuz.stopcorona.skeleton.core.screens.base.viewmodel.ScopedViewModel
+import at.roteskreuz.stopcorona.skeleton.core.utils.observeOnMainThread
+import com.google.android.gms.common.api.ApiException
+import com.google.android.gms.common.api.Status
+import com.google.android.gms.nearby.exposurenotification.ExposureNotificationStatusCodes
+import com.google.android.gms.nearby.exposurenotification.TemporaryExposureKey
 import io.reactivex.Observable
 import io.reactivex.rxkotlin.Observables
+import io.reactivex.rxkotlin.plusAssign
 import kotlinx.coroutines.launch
 import org.threeten.bp.ZonedDateTime
+import timber.log.Timber
 
 /**
  * Handles the user interaction and provides data for [ReportingStatusFragment].
  */
 class ReportingStatusViewModel(
     appDispatchers: AppDispatchers,
     private val reportingRepository: ReportingRepository,
-    private val quarantineRepository: QuarantineRepository
+    private val quarantineRepository: QuarantineRepository,
+    private val exposureNotificationRepository: ExposureNotificationRepository
 ) : ScopedViewModel(appDispatchers) {
-
     private val uploadReportDataStateObserver = DataStateObserver<MessageType>()
+    private val exposureNotificationsErrorState = DataStateObserver<Pair<Status, ExposureNotificationRepository.DebugAction>>()
 
     fun setUserAgreement(agreement: Boolean) {
         reportingRepository.setUserAgreement(agreement)
     }
 
     fun uploadInfectionInformation() {
+        exposureNotificationsErrorState.loading()
+        disposables += exposureNotificationRepository.observeTemporaryExposureKeys()
+            .observeOnMainThread()
```

Comment:
You are in ViewModel. Please don't touch Android framework if possible. Btw. you don't need Main thread here.